### ISSUE TITLE
Nominations: Link previous nomination threads

### ISSUE THREAD

**Issue by bast0006** [python-discord/bot#2304]
It would be nice if we could add links to previous nomination threads to both !tp history and newly created nomination threads. As it stands, we have to search for the username in the discord thread history (and this isn't particularly great UX nor reliable). 

Storing the thread ID for a nomination sounds like a good first step (and probably worth committing immediately even before other features are complete?), then searching through threads by username to attempt to rediscover old threads. Since usernames are changeable, we might also want to detect "desynced" talentpool history, where we know there have been X previous nominations, but only found less-than-X threads. 

We could also search through threads by time--we know when previous nominations have _ended_, but we don't know when they start, so we can use this also to help figure out which threads might be relevant by `archive_timestamp` or `last_pin_timestamp`, the latter of which might actually be extremely reliable (given reviews are auto-pinned in their thread).

**Comment by shtlrs** [comment]
@bast0006 IF you could explain what nominations are & how they work in a Nutshell, I'll gladly take this on once we finish planning it up

**Comment by wookie184** [comment]
> @bast0006 IF you could explain what nominations are & how they work in a Nutshell, I'll gladly take this on once we finish planning it up

The talentpool is our system for getting new helpers (users with the "helper" role), and nominations are a part of this. The overview of how it works is:
1) Any helper can run the `!nominate` command (in a staff channel), to nominate someone they think could be a good helper
2) When a user is nominated they are put in a "pool" of users waiting to be reviewed/voted on
3) The bot chooses a user to put up for vote and sends a message in the `#nomination-voting` voting with contents of the review. A thread is also created from the message, which is where moderators can discuss a user.
4) Moderators+admins vote on the user to decide whether they should be a helper.
5) The vote ends and the vote message is deleted (although threads still exist when the message is deleted).

Sometimes a user doesn't pass the vote but is renominated and there is another vote at a later point in time. In these cases we want to be able to see the context from the old vote so a link to the old thread would be helpful.

I think this should be implemented by storing the thread ID along with the nomination on the site database, as suggested. Then when reviewing a user we can look through their previous nominations and get the thread IDs of them to link.

It would be good to wait for https://github.com/python-discord/bot/pull/2307 to be merged before starting anything on the bot side of things if you're interested in tackling this, to prevent merge conflicts.

**Comment by shtlrs** [comment]
Thanks @wookie184, that's gives me a very good basic understanding of how it works!

Yes, I'll gladly wait on your PR then I'll tackle this if approved!

**Comment by wookie184** [comment]
@shtlrs My PR has been merged so you can start implementing if you're still interested 😃

**Comment by shtlrs** [comment]
Aye aye cap !

**Comment by shtlrs** [comment]
@wookie184 Would adding a `thread_id` column for each nomination entry be considered here?
Just before we edit the nomination [here](https://github.com/python-discord/bot/blob/8faf181012860bd8c60bd386d843af09f472d3fb/bot/exts/recruitment/talentpool/_review.py#L183) after the creating the thread, we attach the newly created thread's id to the nomination entry, which will make retrieval faster.

And then, upon invoking the history, we would add a mention of that thread to the paginated result.

WDYT ?

**Comment by wookie184** [comment]
Sounds good to me!

Previous threads can be added somewhere to the review text in `_previous_nominations_review`, and adding them to the history command too sounds good.

One note, I think the `thread_id` column would be on the "nominations" table instead of the "nomination entry" table, although maybe that's what you meant.

**Comment by shtlrs** [comment]
Here's a preview of what it would look like 

**History command**
![image](https://user-images.githubusercontent.com/48383734/200161233-9aea966e-18ed-4344-a325-2744deddc2eb.png)

**Get review command**

![image](https://user-images.githubusercontent.com/48383734/200161289-bc1e51e3-827c-4b14-984d-a4b69de5daec.png)

**Comment by shtlrs** [comment]
I'm reopening this issue since the most of the code was unfortunately lost due to a merge.
A new PR will be made to reinclude the previously implemented changes [here](https://github.com/python-discord/bot/pull/2319).
